### ISSUE TITLE
Handling different environments

### ISSUE THREAD

**Issue by endel** [doubleleft/hook#134]
Currently, the following files are tied to only ONE environment:

- `hook-ext/config.yaml`
- `hook-ext/security.yaml`
- `.hook-config`

We need a way to handle multiple environments.

**Comment by endel** [comment]
Proposal syntax for `hook-ext/config.yaml`:

```yaml
# hook-ext/config.yaml
mail:
  driver: 'sendmail'

production:
  mail:
    driver: 'smtp'
    host: 'localhost'

staging:
  something: 2
```

The default configuration shouldn't have any environment key, which means it is shared across all environment configurations.

**Comment by endel** [comment]
Fixed in hook-cli v0.3.1: https://github.com/doubleleft/hook-cli/releases/tag/v0.3.1
